libmscore/style.cpp

```diff
@@ -2474,18 +2534,221 @@ void MStyle::save(XmlWriter& xml, bool optimize)
             _chordList.write(xml);
             xml.etag();
             }
-      xml.tag("Spatium", value(Sid::spatium).toDouble() / DPMM);
+      xml.tag("Spatium", value(Sid::spatium).toDouble() / DPMM);                 // 3.01
+      xml.tag("spatium", rint(value(Sid::spatium).toDouble() / DPI_F * MSCX_F)); // 3.01+
       xml.etag();
       }
 
+//--------------------------------------------------------------------------
+//   initPageLayout - only applied to MScore::_baseStyle and _defaultStyle
+//                    must wait until setMscoreLocale() is called in main()
+//--------------------------------------------------------------------------
+
+void MStyle::initPageLayout()
+      { // by default, units are millimeters or inches, depending on locale
+        // default preferences are initialized prior to default styles in main()
+      QPageLayout::Unit unit;
+      QPageSize::PageSizeId psid;
+      unit = QPageLayout::Unit(MScore::unitsValue());
+      psid = MStyle::isMetric(unit) ? QPageSize::A4 : QPageSize::Letter;
+      _pageSize = QPageSize(psid);
+      _pageOdd  = MPageLayout();
+      _pageOdd.setPageSize(_pageSize);
+      _pageOdd.setUnits(QPageLayout::Millimeter);
+      _pageOdd.setMargins(QMarginsF(10, 10, 10, 20)); // default margins in millimeters
+cout << "initPageLayout - Odd Left Margin: " << _pageOdd.margins(QPageLayout::Inch).left() << endl;
+      _pageOdd.setUnits(unit);                        // set to user locale units
+      _pageOdd.setOrientation(QPageLayout::Portrait);
+      _pageEven = MPageLayout(_pageOdd);
+
+      set(Sid::pageUnits, int(unit));
+      set(Sid::pageSize,  int(psid));
+
+      fromPageLayout(); // sync the styles
+      }
+
+//------------------------------------------------------------------------------
+//   from & toPageLayout convert between QPageLayout/QPageSize and style values
+//------------------------------------------------------------------------------
+void MStyle::fromPageLayout()
+      {
+cout << "fromPageLayout - Odd Left Margin: " << _pageOdd.margins(QPageLayout::Inch).left() << endl;
```

Comment:
Use `qDebug()` instead of `cout` to avoid excessive output in release versions... And spacing is wrong.

Reply:
As explained in a comment on the PR page, these are temporary and purely for the purpose of illustrating the Travis issue that I can only work around.  The spacing is on purpose to call them out as temporary.